File: 0350-intersection-of-two-arrays-ii/0350-intersection-of-two-arrays-ii.java
class Solution {
    public int[] intersect(int[] nums1, int[] nums2) {
        int[] cnt = new int[1001];
        int[] res = new int[1001];

        for (int n: nums1) cnt[n]++;

        int idx = 0;
        for (int n: nums2) if (cnt[n]-- > 0) res[idx++] = n;

        return Arrays.copyOfRange(res, 0, idx);
    }
}

File: 0350-intersection-of-two-arrays-ii/0350-intersection-of-two-arrays-ii.cpp
class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        int cnt[1001] = { 0 };
        vector<int> res;

        for (int n: nums1) cnt[n]++;

        for (int n: nums2) if (cnt[n]-- > 0) res.push_back(n);

        return res;
    }
};

auto init = []() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    cout.tie(nullptr);
    return 'c';
}();